Advanced Tutorial.md

```diff
@@ -0,0 +1,211 @@
+# Advanced API Blueprint Tutorial
+
+Welcome to the advanced API Blueprint tutorial! This tutorial will take you through advanced topics like JSON Schema, request and response attributes, data structures and relation types.
+
+This tutorial assumes that you have read the [API Blueprint Tutorial](./Tutorial.md).
+
+## JSON Schema
+
+Action request and response bodies can have associated schemas that describe the allowed structure of the body content. JSON bodies are typically described with [JSON Schema](http://json-schema.org/). Given a simple JSON response body we can describe the structure of the response with JSON Schema in a `+ Schema` section:
+
+```apib
+## Random Numbers [/random]
+
+### Generate a random number [GET]
+Generates and returns a random number.
+
++ Response 200 (application/json)
+
+    + Body
+
+            {
+              "value": 12345
+            }
+
+    + Schema
+
+            {
+              "type": "object",
+              "properties": {
+                "value": {
+                  "type": "number"
+                }
+              }
+            }
+```
+
+The schema can describe the type of each member, which members are required, default values, and support a number of other advanced features. Below is a more complete example, taken from the API Blueprint tutorial blueprint:
+
+```apib
+### List All Questions [GET]
++ Response 200 (application/json)
+
+    + Body
+
+            [
+              {
+                "question": "Favourite programming language?",
+                "published_at": "2014-11-11T08:40:51.620Z",
+                "url": "/questions/1",
+                "choices": [
+                  {
+                    "choice": "Swift",
+                    "url": "/questions/1/choices/1",
+                    "votes": 2048
+                  }, {
+                    "choice": "Python",
+                    "url": "/questions/1/choices/2",
+                    "votes": 1024
+                  }, {
+                    "choice": "Objective-C",
+                    "url": "/questions/1/choices/3",
+                    "votes": 512
+                  }, {
+                    "choice": "Ruby",
+                    "url": "/questions/1/choices/4",
+                    "votes": 256
+                  }
+                ]
+              }
+            ]
+
+    + Schema
+
+            {
```

Comment:
This tutorial produces a large JSON Schema, I wonder if we would be better off describing a simpler structure, perhaps the structure used on creating a question instead:

```apib
+ Request (application/json)
    + Body

            {
              "question": "Favourite language?"
              "choices": [ "Swift", "Objective-C" ]
            }

    + Schema

            {
              "$schema": "http://json-schema.org/draft-04/schema#",
              "type": "object",
              "properties": {
                "question": { "type": "string }
                "choices": {
                  "type": "array",
                  "items": { "type": "string" },
                  "minItems": 2
                }
              }
            }
```